core/src/main/java/org/apache/iceberg/MergingSnapshotUpdate.java

```diff
@@ -198,9 +197,7 @@ protected void add(DataFile file) {
     try {
       if (newFiles.size() > 0) {
         // add all of the new files to the summary builder
-        for (DataFile file : newFiles) {
-          summaryBuilder.addedFile(spec, file);
-        }
+        newFiles.forEach(file -> summaryBuilder.addedFile(spec, file));
```

Comment:
Does this get optimized out by the compiler?

Reply:
Not sure what you mean - can you clarify?

Reply:
Just wondering if there's a penalty for changing iteration from a loop to a lambda call. If these are equivalent in byte code, then we can just pick the style we prefer. If there are times when the labmda is going to be slower, then it makes sense to continue using the old for syntax.

I'm also reluctant to make this change because it seems like it is mostly churn. The two are equivalent, why change existing code and touch more lines with this commit?